### ISSUE TITLE
Three nearly identical Dirichlet boundary loss functions

### ISSUE THREAD

**Issue by zeinali72** [zeinali72/swe-pinn#57]
## Description

In `src/losses/boundary.py`, `loss_boundary_dirichlet_h`, `loss_boundary_dirichlet_hu`, and `loss_boundary_dirichlet_hv` (lines 8-38) are identical except for the output index (0, 1, 2).

## Fix

Consolidate into one parameterised function:

```python
def loss_boundary_dirichlet(model, params, batch, target, var_idx):
    U_pred = model.apply({'params': params['params']}, batch, train=False)
    pred = U_pred[..., var_idx]
    if target.ndim != pred.ndim:
        target = target.reshape(pred.shape)
    return jnp.mean((pred - target)**2)
```

Keep the three original names as thin wrappers if needed for backward compatibility.

## Category
**Medium** · Pillar: Redundancy Check
